neurodamus/utils/memory.py

```diff
@@ -459,54 +474,62 @@ def get_num_target_ranks(self, num_ranks):
             return int(num_ranks)
 
     @run_only_rank0
-    def distribute_cells(self, num_ranks, batch_size=10) -> (dict, dict):
+    def distribute_cells(self, num_ranks, cycles=None, batch_size=10) -> (dict, dict, dict):
         """
-        Distributes cells across ranks based on their memory load.
+        Distributes cells across ranks and cycles based on their memory load.
 
-        This function uses a greedy algorithm to distribute cells across ranks such that
+        This function uses a greedy algorithm to distribute cells across ranks and cycles such that
         the total memory load is balanced. Cells with higher memory load are distributed first.
 
         Args:
-            dry_run_stats (DryRunStats): A DryRunStats object.
             num_ranks (int): The number of ranks.
+            cycles (int): The number of cycles to distribute cells over.
+            batch_size (int): The number of cells to assign to each bucket at a time.
 
         Returns:
-            rank_allocation (dict): A dictionary where keys are rank IDs and
-                                    values are lists of cell IDs assigned to each rank.
-            rank_memory (dict): A dictionary where keys are rank IDs
-                                and values are the total memory load on each rank.
+            bucket_allocation (dict): A dictionary where keys are tuples (pop, rank_id, cycle_id)
+                                    and values are lists of cell IDs assigned to each bucket.
+            bucket_memory (dict): A dictionary where keys are tuples (pop, rank_id, cycle_id)
+                                and values are the total memory load on each bucket.
+            metype_memory_usage (dict): A dictionary where keys are METype IDs
+                                        and values are the memory load of each METype.
         """
-        logging.info("Distributing cells across %d ranks", num_ranks)
+        if cycles is None or cycles <= 1:
```

Comment:
```suggestion
        if not cycles:
```
unless there's a real chance that someone will use negative numbers here.